amd_openvx_extensions/amd_rpp/source/Brightness.cpp

```diff
@@ -21,100 +21,102 @@ THE SOFTWARE.
-{
+struct BrightnessLocalData {
     RPPCommonHandle handle;
     rppHandle_t rppHandle;
-    Rpp32u device_type;
+    Rpp32u deviceType;
     Rpp32u nbatchSize;
     RppPtr_t pSrc;
     RppPtr_t pDst;
     vx_float32 *alpha;
     vx_float32 *beta;
-    RpptDescPtr src_desc_ptr;
+    RpptDescPtr srcDescPtr;
     RpptDesc srcDesc;
     RpptDesc dstDesc;
-    RpptDescPtr dst_desc_ptr;
-    RpptROI *roi_tensor_ptr;
+    RpptDescPtr dstDescPtr;
+    void *roiTensorPtr;
+    RpptROI *roiPtr;
     RpptRoiType roiType;
-    Rpp32u layout;
-    size_t in_tensor_dims[NUM_OF_DIMS];
-    size_t out_tensor_dims[NUM_OF_DIMS];
-    vx_enum in_tensor_type;
-    vx_enum out_tensor_type;
+    Rpp32s inputLayout;
+    Rpp32s outputLayout;
+    size_t inputTensorDims[NUM_OF_DIMS];
+    size_t ouputTensorDims[NUM_OF_DIMS];
+    vx_enum inputTensorType;
+    vx_enum outputTensorType;
 #if ENABLE_HIP
     void *pSrc_dev;
     void *pDst_dev;
-    RpptROI *roi_tensor_ptr_dev;
 #endif
 };
 
-static vx_status VX_CALLBACK refreshBrightness(vx_node node, const vx_reference *parameters, vx_uint32 num, BrightnessLocalData *data)
-{
+static vx_status VX_CALLBACK refreshBrightness(vx_node node, const vx_reference *parameters, vx_uint32 num, BrightnessLocalData *data) {
     vx_status status = VX_SUCCESS;
-    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[1], 0, data->nbatchSize * 4, sizeof(unsigned), data->roi_tensor_ptr, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
     STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[3], 0, data->nbatchSize, sizeof(vx_float32), data->alpha, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
     STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[4], 0, data->nbatchSize, sizeof(vx_float32), data->beta, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
-    
-    if(data->layout == 2 || data->layout == 3) // For NFCHW and NFHWC formats
-    {
-        unsigned num_of_frames = data->in_tensor_dims[1]; // Num of frames 'F'
-        for(int n = data->nbatchSize - 1; n >= 0; n--)
-        {
-            unsigned index = n * num_of_frames;
-            for(int f = 0; f < num_of_frames; f++)
-            {
-                data->alpha[index + f] = data->alpha[n];
-                data->beta[index + f] = data->beta[n];
-                data->roi_tensor_ptr[index + f].xywhROI.xy.x = data->roi_tensor_ptr[n].xywhROI.xy.x;
-                data->roi_tensor_ptr[index + f].xywhROI.xy.y = data->roi_tensor_ptr[n].xywhROI.xy.y;
-                data->roi_tensor_ptr[index + f].xywhROI.roiWidth = data->roi_tensor_ptr[n].xywhROI.roiWidth;
-                data->roi_tensor_ptr[index + f].xywhROI.roiHeight = data->roi_tensor_ptr[n].xywhROI.roiHeight;
-            }
-        }
-    }
-    if (data->device_type == AGO_TARGET_AFFINITY_GPU)
-    {
+
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
 #if ENABLE_HIP
+        STATUS_ERROR_CHECK(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HIP, &data->roiTensorPtr, sizeof(data->roiTensorPtr)));
```

Comment:
roiTensorPtr is a void pointer. Please check if sizeof(data->roiTensorPtr) works as expected

Reply:
Generally that is how even the src and dst pointers are queried please check the lines 61 and 62